main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -302,6 +302,49 @@ object Verifier {
             case _ => failMismatchedShape(t1, "Tuple", loc)
           }
 
+        case AtomicOp.Assign =>
+          val List(t1, t2) = ts
+          t1 match {
+            case MonoType.Ref(elm) =>
+              checkEq(t2, elm, loc)
+              check(expected = MonoType.Unit)(actual = tpe, loc)
+            case _ => failMismatchedShape(t1, "Ref", loc)
+          }
+
+        // Match- and Hole-errors match with any type
+        case AtomicOp.HoleError(_) | AtomicOp.MatchError => tpe
```

Comment:
For pattern matches we usually format everything consistently, so a single case per case so to speak, and newline after `=>` and before `case`